README.md

```diff
@@ -1,85 +1,155 @@
-| `/question/edit`               | POST   | Used to edit question                               | `id` <br> `title` <br> `complexity` <br> `description` <br> `tags`    | Yes                                | Yes                              |
-| `/question/getAll`             | GET    | Used to get all the questions from the database     | -                                                                     | Yes                                | No                               |
-| `/question/getQuestionDetails` | GET    | Used to get the details of the specified question   | `id`                                                                  | Yes                                | No                               |
-| `/user/change-password`        | POST   | Used to change user password                        | `id` <br> `currentPassword` <br> `newPassword` <br> `confirmPassword` | Yes                                | No                               |
-| `/user/delete`                 | POST   | Used to delete user                                 | `id`                                                                  | Yes                                | No                               |
-| `/user/login`                  | POST   | Used to login                                       | `email` <br> `password`                                               | No                                 | -                                |
-| `/user/read`                   | POST   | Used to get user information                        | `id` or `email`                                                       | Yes                                | No                               |
-| `/user/readAll`                | GET    | Used to get all users information                   | -                                                                     | Yes                                | Yes                              |
-| `/user/signup`                 | POST   | Used to create new user                             | `email` <br> `password` <br> `confirmPassword`                        | No                                 | -                                |
-| `/user/update`                 | POST   | Used to update user information (username)          | `id` <br> `username`                                                  | Yes                                | No                               |
-
-- `auth` API (port 5001) contains all the authorization related endpoints.
-- `question` API (port 3001) contains all the question data related endpoints.
-- `user` API (port 4001) contains all the user data related endpoints (including authentication).
-- Note that if the API path requires JWT token to be in the header, it means the user has to be logged in.
```

Comment:
I think we can include this in for the marker to test that the microservices are working without using the frontend if they want to just test the backend only

Reply:
Added back as a separate `API_Endpoints.md`, which is linked to `README.md`.